### PR TITLE
[#44][Backend] Add unit tests for results controller

### LINKED ISSUE #44: [Chore] Add unit tests for `results_controller`
## Why

Currently, we don't have tests for `results_controller`, as a developer, we need to improve the code coverage of the controller to 100%

## Acceptance Criteria

- Add unit tests for list results, and also can filter by keyword
- Add unit tests for result detail

### PR DESCRIPTION
**Pull request by sanG-github**
Resolve #44 

## Why

Currently, we don't have tests for `results_controller`, as a developer, we need to improve the code coverage of the controller to 100%

## Acceptance Criteria

- Add unit tests for list results in case filtered by keyword or not
- Add unit tests for result detail

## Proof of Work

Screen Shot 2022-10-13 at 16 41 13

### CODE REVIEW

File: spec/controllers/results_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    context 'when filtered by keyword' do
+      it 'returns all filtered results of user' do
+        user = create :user
+        other_user = create :user
+        keyword = 'e'
+        attachment = create :attachment, user_id: user.id
+        nimble_result = create :result, keyword: 'Nimble', attachment_id: attachment.id
+        company_result = create :result, keyword: 'company', attachment_id: attachment.id
+        vietnam_result = create :result, keyword: 'VietNam', attachment_id: attachment.id
+        un_authored_attachment = create :attachment, user_id: other_user.id
```
Comment (rosle) [inline spec/controllers/results_controller_spec.rb:38]: Maybe we should create a result with the keyword for this attachment containing `e`, so we're sure that the other user's keyword result with `e` is not returned. 💡

File: spec/controllers/results_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  end
+
+  describe 'GET #index' do
+    context 'without params' do
+      it 'returns all results of user' do
+        user = create :user
+        other_user = create :user
+        number_of_results = 10
+        attachment = create :attachment, user_id: user.id
+        un_authored_attachment = create :attachment, user_id: other_user.id
```
Comment (rosle) [inline spec/controllers/results_controller_spec.rb:15]: Should we create the result for this attachment 🤔 💭

File: spec/controllers/results_controller_spec.rb (hunk 1 above)
Comment (rosle) [inline spec/controllers/results_controller_spec.rb:38]: I'm thinking of renaming this variable to 

```suggestion
        another_user_attachment = create :attachment, user_id: other_user.id
```

But not sure, do you think it is clearer? 🤔 💭

Comment (sanG-github) [reply spec/controllers/results_controller_spec.rb]: Resolved it in e8767c3 ✅

File: spec/controllers/results_controller_spec.rb (hunk 2 above)
Comment (sanG-github) [reply spec/controllers/results_controller_spec.rb]: Resolved it in d37517e ✅

File: spec/controllers/results_controller_spec.rb (hunk 1 above)
Comment (sanG-github) [reply spec/controllers/results_controller_spec.rb]: Resolved it in d37517e ✅

File: spec/controllers/results_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+      sign_in user
+      get :show, params: params
+
+      expect(response).to have_http_status :success
+      expect(controller.instance_variable_get(:@result)).to be_a(Result)
+      expect(controller.instance_variable_get(:@result)).to eq(result)
+    end
+
+    it 'returns nil for un-authored result' do
```
[the remark this answers is not in the document]
Comment (sanG-github) [reply spec/controllers/results_controller_spec.rb]: Yep, I've updated it to raise an error instead in d37517e ✅

**Review by tyrro — APPROVED** [review]

**Review by byhbt — APPROVED** [review]

**Pull request merged**